legacy_packages/wallets/src/evm/connectors/smart-wallet/lib/base-api.ts

```diff
@@ -333,6 +333,9 @@ export abstract class BaseAccountAPI {
       );
     }
 
+    // add 50k overhead for entrypoint checks (following same as unity-sdk)
+    partialOp.callGasLimit = partialOp.callGasLimit.add(50_000);
```

Comment:
you need to do this in diff places depending on the path here.

line 284
line 299
line 327